hazelcast-sql/src/test/java/com/hazelcast/sql/misc/SqlNoSerializationTest.java

```diff
@@ -54,66 +66,61 @@
 @RunWith(Parameterized.class)
 @Parameterized.UseParametersRunnerFactory(HazelcastSerialParametersRunnerFactory.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
-public class SqlNoSerializationTest extends SqlTestSupport {
+public class SqlNoSerializationTest extends OptimizerTestSupport {
 
     private static final String MAP_NAME = "map";
     private static final int KEY_COUNT = 100;
 
     @Parameterized.Parameter
     public boolean useIndex;
 
-    private final TestHazelcastFactory factory = new TestHazelcastFactory();
-    private HazelcastInstance member;
-
     private static volatile boolean failOnSerialization;
 
     @Parameterized.Parameters(name = "useIndex:{0}")
     public static Collection<Object> parameters() {
-        return Arrays.asList(true, false);
+        return Arrays.asList(false, true);
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        initialize(1, smallInstanceConfig());
     }
 
     @Before
     public void before() {
-        member = factory.newHazelcastInstance();
-        factory.newHazelcastInstance();
-
         Map<Key, Value> localMap = new HashMap<>();
 
         for (int i = 0; i < KEY_COUNT; i++) {
             localMap.put(new Key(i), new Value(i));
         }
 
-        IMap<Key, Value> map = member.getMap(MAP_NAME);
+        IMap<Key, Value> map = instance().getMap(MAP_NAME);
 
         map.putAll(localMap);
 
         // An index may change the behavior due to MapContainer.isUseCachedDeserializedValuesEnabled.
         if (useIndex) {
-            map.addIndex(new IndexConfig().setType(IndexType.SORTED).addAttribute("val"));
+            map.addIndex(new IndexConfig(IndexType.HASH, "val"));
         }
 
         failOnSerialization = true;
     }
 
     @After
     public void after() {
+        instance().getMap(MAP_NAME).clear();
         failOnSerialization = false;
-
-        factory.shutdownAll();
     }
 
     @Test
-    public void testMapScan() {
```

Comment:
Why we don't want to test _regular_ scan anymore?

Reply:
Actually, we [test both scan types](https://github.com/hazelcast/hazelcast/pull/19162/files/ab1ee47050861f94f7c77904c9f4f4dfe47b72d4#diff-4f62ce3a66b37bf9670d6c9858c3ffde09d265a10bbb1750c4d7a90c315b2740R153-R167).